src/Subscribers/DoctrineEncryptSubscriber.php

```diff
@@ -146,16 +150,39 @@ public function postLoad(LifecycleEventArgs $args)
     }
 
     /**
-     * Listen to preflush event
+     * Listen to onflush event
      * Encrypt entities that are inserted into the database
      *
      * @param PreFlushEventArgs $preFlushEventArgs
      */
     public function preFlush(PreFlushEventArgs $preFlushEventArgs)
     {
-        $unitOfWork = $preFlushEventArgs->getEntityManager()->getUnitOfWork();
+        $unitOfWOrk = $preFlushEventArgs->getEntityManager()->getUnitOfWork();
+        foreach ($unitOfWOrk->getIdentityMap() as $entityName => $entityArray) {
+            if (isset($this->cachedDecryptions[$entityName])) {
+                foreach ($entityArray as $entityId => $instance) {
+                    $this->processFields($instance);
+                }
+            }
+        }
+    }
+
+    /**
+     * Listen to onflush event
+     * Encrypt entities that are inserted into the database
+     *
+     * @param OnFlushEventArgs $onFlushEventArgs
+     */
+    public function onFlush(OnFlushEventArgs $onFlushEventArgs)
```

Comment:
what exactly is the reason to do onFlush? are cascading inserts only seen here, but not in preFlush? preFlush would see ALL updates and all new entities where we called `persist` explicitly, but not entities with cascade persist?

Reply:
That's right. A new entity on an association with an existing entity that has no explicit `persist` called on it is only added to the Unit Of Work after the `preFlush` event.